Sources/XiEditor/ClientImplementation.swift

```diff
@@ -182,30 +179,34 @@ class ClientImplementation: XiClient, DocumentsProviding, ConfigCacheProviding,
 
     // Stores the config dict so new windows don't have to wait for core to send it.
     // The main purpose of this is ensuring that `unified_titlebar` applies immediately.
-    var configCache: [String: AnyObject] = [:]
+    var configCache = Config(fontFace: nil, fontSize: nil, scrollPastEnd: nil, unifiedToolbar: nil)
+
+    func configChanged(viewIdentifier: ViewIdentifier, changes: Config) {
+        self.configCache = Config(
+            fontFace: changes.fontFace ?? self.configCache.fontFace,
+            fontSize: changes.fontSize ?? self.configCache.fontSize,
+            scrollPastEnd: changes.scrollPastEnd ?? self.configCache.scrollPastEnd,
+            unifiedToolbar: changes.unifiedToolbar ?? self.configCache.unifiedToolbar
+        )
 
-    func configChanged(viewIdentifier: ViewIdentifier, changes: [String : AnyObject]) {
-        for (key, value) in changes {
-            self.configCache[key] = value
-        }
         let document = documentForViewIdentifier(viewIdentifier: viewIdentifier)
         DispatchQueue.main.async {
-            document?.editViewController?.configChanged(changes: changes)
+            document?.editViewController?.configChanged(config: self.configCache)
         }
     }
 
-    func measureWidth(args: [[String : AnyObject]]) -> [[Double]] {
+    func measureWidth(args: [MeasureWidthParams]) -> [[Double]] {
```

Comment:
We still mix and match between the terms `Params` and `Args`, but I think we can do a sweep of cleaning that up later if anyone feels strongly about one or the other. :)

Reply:
yea, this is a 🤷‍♂ for me. 'params' is used in the JSON-RPC spec, so that's why we use that term widely; and then 'args' is probably just used out of force of habit. If we were to pick one I'd pick params.